src/main/java/org/templegalaxia/model/Petal.java

```diff
@@ -4,58 +4,20 @@
 import heronarts.lx.model.LXModel;
 import heronarts.lx.model.LXPoint;
 import heronarts.lx.transform.LXTransform;
-import java.io.BufferedReader;
-import java.io.InputStreamReader;
-import java.util.ArrayList;
-import java.util.List;
-import org.apache.commons.csv.CSVFormat;
-import org.apache.commons.csv.CSVParser;
-import org.apache.commons.csv.CSVRecord;
 
 public class Petal extends LXModel {
-  public static final int numPixels;
 
-  public Petal(LXTransform transform) {
-    super(new Fixture(transform));
-  }
-
-  private static List<float[]> xyzs = new ArrayList<>();
-
-  // Statically initialize the CSV load on the first init
-  static {
-    if (xyzs.size() == 0) {
-      CSVParser parser;
-      try {
-        parser =
-            new CSVParser(
-                new BufferedReader(
-                    new InputStreamReader(Petal.class.getResourceAsStream("petalPoints.csv"))),
-                CSVFormat.EXCEL.withHeader());
-      } catch (Exception e) {
-        // Coherce to non checked exceptions.
-        throw new RuntimeException(e);
-      }
+  private static PointLoader pointArray = new PointLoader("petalPoints.csv");
```

Comment:
I would call this `pointLoader` rather than `pointArray`, since it's not an array.